@jondubois Also, it's failing on Jenkins with
```
@liskhq/lisk-p2p: test/unit/peer_discovery.ts(54,52): error TS2345: Argument of type 'P2PPeerInfo[]' is not assignable to parameter of type 'ReadonlyArray<P2PDiscoveredPeerInfo>'.

@liskhq/lisk-p2p:   Property 'version' is missing in type 'P2PPeerInfo' but required in type 'P2PDiscoveredPeerInfo'.
```

--- a/packages/lisk-p2p/src/p2p.ts
+++ b/packages/lisk-p2p/src/p2p.ts
@@ -389,7 +389,8 @@ export class P2P extends EventEmitter {
 			this._config.blacklistedPeers,
 		);
 
-		// Make sure that we do not try to connect to peers if the P2P node is no longer active.
+		// Stop discovery if node is no longer active. That way we don't try to connect to peers.
+		// We need to check again because of the previous asynchronous await statement.
 		if (!this._isActive) {
 			
 			return;

--- a/packages/lisk-p2p/test/unit/peer.ts
+++ b/packages/lisk-p2p/test/unit/peer.ts
@@ -24,6 +24,7 @@ describe('peer', () => {
 		height: 545776,
 		os: 'darwin',
 		version: '1.1.0',
+		isDiscoveredPeer: true
 	};
 
 	const defaultPeer = new Peer(defaultPeerInfo);

--- a/packages/lisk-p2p/test/unit/peer_discovery.ts
+++ b/packages/lisk-p2p/test/unit/peer_discovery.ts
@@ -13,7 +13,7 @@
  *
  */
 import { expect } from 'chai';
-import { P2PPeerInfo } from '../../src/p2p_types';
+import { P2PPeerInfo, P2PDiscoveredPeerInfo } from '../../src/p2p_types';
 import { initializePeerList } from '../utils/peers';
 import * as discoverPeersModule from '../../src/peer_discovery';
 
@@ -23,22 +23,31 @@ describe('peer discovery', () => {
 	const seedPeer2 = samplePeers[1];
 	const seedList = [seedPeer1, seedPeer2];
 
-	const validatedPeer1: P2PPeerInfo = {
+	const validatedPeer1: P2PDiscoveredPeerInfo = {
 		ipAddress: '196.34.89.90',
 		wsPort: 5393,
 		height: 23232,
+		os: 'linux',
+		version: '1.0.0',
+		isDiscoveredPeer: true,
 	};
 
-	const validatedPeer2: P2PPeerInfo = {
+	const validatedPeer2: P2PDiscoveredPeerInfo = {
 		ipAddress: '128.38.75.9',
 		wsPort: 5393,
 		height: 23232,
+		os: 'linux',
+		version: '1.0.0',
+		isDiscoveredPeer: true,
 	};
 
-	const validatedPeer3: P2PPeerInfo = {
+	const validatedPeer3: P2PDiscoveredPeerInfo = {
 		ipAddress: '12.23.11.31',
 		wsPort: 5393,
 		height: 23232,
+		os: 'linux',
+		version: '1.0.0',
+		isDiscoveredPeer: true,
 	};
 
 	describe('#discoverPeer', () => {

